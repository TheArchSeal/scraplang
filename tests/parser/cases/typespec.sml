var x = 0;
var x: u8 = 0;
var x: u8* = 0;
var x: u8[] = 0;
var x: u8*[]* = 0;
var x: () => u8 = 0;
var x: (u8) => u8* = 0;
var x: ((u8, u8?, u8[]?) => () => u8)* = 0;
