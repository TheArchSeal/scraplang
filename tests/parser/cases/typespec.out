stmt (4):1:1 var x
    expr (2):1:9 0
    type (1):1:5 (inferred)
stmt (4):2:1 var x
    expr (2):2:13 0
    type (2):2:8 u8
stmt (4):3:1 var x
    expr (2):3:14 0
    type (4):3:8 *
        type (2):3:8 u8
stmt (4):4:1 var x
    expr (2):4:15 0
    type (3):4:8 []
        type (2):4:8 u8
stmt (4):5:1 var x
    expr (2):5:17 0
    type (4):5:8 *
        type (3):5:8 []
            type (4):5:8 *
                type (2):5:8 u8
stmt (4):6:1 var x
    expr (2):6:19 0
    type (5):6:8 (0?)=>
        type (2):6:14 u8
stmt (4):7:1 var x
    expr (2):7:22 0
    type (5):7:8 (0?)=>
        type (2):7:9 u8
        type (4):7:16 *
            type (2):7:16 u8
stmt (4):8:1 var x
    expr (2):8:47 0
    type (4):8:8 const*
        type (5):8:8 (2?)=>
            type (2):8:10 u8
            type (2):8:14 u8
            type (3):8:19 []
                type (2):8:19 u8
            type (5):8:29 (0?)=>
                type (2):8:35 u8
